lib/vtls/schannel_verify.c

```diff
@@ -76,6 +77,20 @@ struct cert_chain_engine_config_win7 {
   HCERTSTORE hExclusiveTrustedPeople;
 };
 
+/*
+* size of the structure: 20 bytes.
+* 12 bytes remain unused when dealing with IPv4 addresses,
+* while IPv6 addresses utilize the entire allotted space.
+*/
+
+struct num_ip_data {
+  DWORD cbData; /* 04 bytes */
```

Comment:
can we make this a plain `int` instead? Just for us not used to funny Microsoft typedefs.

Also, how about calling it `size`, `iplen` or something instead of `cbData` to better explain what it is for?

Reply:
How about a compromise? Let's rename the `cbData` to `size`, and call it good. Because later in the code there is a bunch of comparisons like this:
```C
/* I've already changed the cbData to size */
if(entry->IPAddress.cbData == p->size) {
```
And here is the definition of the `CERT_ALT_NAME_ENTRY` (entry) and `IPAddress`:

```C
typedef struct _CERT_ALT_NAME_ENTRY {
    DWORD   dwAltNameChoice;
    union {                                             // certenrolls_skip
        PCERT_OTHER_NAME            pOtherName;         // 1
        LPWSTR                      pwszRfc822Name;     // 2  (encoded IA5)
        LPWSTR                      pwszDNSName;        // 3  (encoded IA5)
        // Not implemented          x400Address;        // 4
        CERT_NAME_BLOB              DirectoryName;      // 5
        // Not implemented          pEdiPartyName;      // 6
        LPWSTR                      pwszURL;            // 7  (encoded IA5)
        CRYPT_DATA_BLOB             IPAddress;          // 8  (Octet String)
        LPSTR                       pszRegisteredID;    // 9  (Object Identifer)
    } DUMMYUNIONNAME;                                   // certenrolls_skip
} CERT_ALT_NAME_ENTRY, *PCERT_ALT_NAME_ENTRY;
```
```C
typedef struct _CRYPTOAPI_BLOB {
                            DWORD   cbData;
    _Field_size_bytes_(cbData)  BYTE    *pbData;
} CRYPT_INTEGER_BLOB, *PCRYPT_INTEGER_BLOB,
CRYPT_UINT_BLOB, *PCRYPT_UINT_BLOB,
CRYPT_OBJID_BLOB, *PCRYPT_OBJID_BLOB,
CERT_NAME_BLOB, *PCERT_NAME_BLOB,
CERT_RDN_VALUE_BLOB, *PCERT_RDN_VALUE_BLOB,
CERT_BLOB, *PCERT_BLOB,
CRL_BLOB, *PCRL_BLOB,
DATA_BLOB, *PDATA_BLOB,
CRYPT_DATA_BLOB, *PCRYPT_DATA_BLOB,
CRYPT_HASH_BLOB, *PCRYPT_HASH_BLOB,
CRYPT_DIGEST_BLOB, *PCRYPT_DIGEST_BLOB,
CRYPT_DER_BLOB, *PCRYPT_DER_BLOB,
CRYPT_ATTR_BLOB, *PCRYPT_ATTR_BLOB;
```
As you can see the `cbData` is defined as `DWORD`. I'm pretty sure that if I change the type to `int`, some stupid ass compiler, let's call it mingw, will start complaining about the comparison between a DWORD and an integer. Let's not rock the boat, and leave the type as is.